Murtaza/accordion/src/components/Accordion.tsx

```diff
@@ -0,0 +1,97 @@
+import {
+  StyleSheet,
+  Text,
+  TouchableOpacity,
+  View,
+  Dimensions,
+} from "react-native";
+import React, { useState, useEffect } from "react";
+import { Entypo } from "@expo/vector-icons";
+import AccordionItem from "./AccordionItem";
+import Animated, {
+  useAnimatedStyle,
+  useSharedValue,
+  withTiming,
+  Easing,
+} from "react-native-reanimated";
+
+interface AccordionProps {
+  title: string;
+  items: string[];
+  isOpen: boolean;
+  onPress: () => void;
+  searchText: string;
+}
+
+const Accordion: React.FC<AccordionProps> = ({
+  title,
+  items,
+  isOpen,
+  onPress,
+  searchText,
+}) => {
+  const rotateZ = useSharedValue(isOpen ? -180 : 0);
+
+  useEffect(() => {
+    rotateZ.value = withTiming(isOpen ? -180 : 0, {
+      duration: 350,
+      easing: Easing.inOut(Easing.quad),
+    });
+  }, [isOpen]);
+
+  const handleAccordionPress = () => {
+    onPress();
+  };
+
+  const animatedStyle = useAnimatedStyle(() => ({
+    transform: [{ rotateZ: `${rotateZ.value}` + "deg" }],
+  }));
+
+  return (
+    <View>
+      <TouchableOpacity style={styles.container} onPress={handleAccordionPress}>
+        <Text style={styles.accordionTitle}>{title}</Text>
+        <Animated.View style={animatedStyle}>
+          <Entypo name="chevron-down" size={24} color="black" />
+        </Animated.View>
+      </TouchableOpacity>
+      <View style={styles.line}></View>
+
+      {isOpen && (
+        <View>
+          {items.map((item, index) => (
```

Comment:
Use Flashlist to improve performance. generally in performance comparison : Flashlist>Flatlist>Map